src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class MethodBuilderImpl : MethodBuilder
+    {
+        private readonly Type _returnType;
+        private readonly Type[]? _parameterTypes;
+        private readonly ModuleBuilderImpl _module;
+
+        internal MethodBuilderImpl(string name, MethodAttributes attributes, CallingConventions callingConventions, Type? returnType,
+            Type[]? parameterTypes, ModuleBuilderImpl module, TypeBuilderImpl declaringType)
+        {
+            _module = module;
+            _returnType = returnType ?? typeof(void);
+            Name = name;
+            Attributes = attributes;
+            CallingConvention = callingConventions;
+            DeclaringType = declaringType;
+            Module = declaringType.Module;
+
+            if (parameterTypes != null)
+            {
+                _parameterTypes = new Type[parameterTypes.Length];
```

Comment:
Validate that the parameters are non-null?

Matching runtime builder code: https://github.com/dotnet/runtime/blob/0b47a4dc519ada9c56d6f4d62865c2d96a56510c/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeMethodBuilder.cs#L73-L76